### PR TITLE
Fix log send for RN IOS only

### PR DESCRIPTION
**Pull request by chrisnojima**
Fixes log send for ios (android coming soon in a different branch off of this)

### CODE REVIEW

File: go/bind/keybase.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func (s serviceCn) NewCrypto(config libkbfs.Config, params libkbfs.InitParams, c
 // LogSend sends a log to Keybase
 func LogSend(uiLogPath string) (string, error) {
 	logSendContext.Logs.Desktop = uiLogPath
-	return logSendContext.LogSend("", 10000)
+	return logSendContext.LogSend("", 5*1024*1024)
```
Comment (chrisnojima) [inline go/bind/keybase.go]: Larger logs

File: shared/dev/log-send/render.native.js [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
-}
-
-class LogSendRender extends Component<void, Props, State> {
-  state: State;
+import type {Props} from './render'
 
-  constructor (props: Props) {
-    super(props)
-    this.state = {copiedToClipboard: false}
+class LogSendRender extends Component<void, Props, {copiedToClipboard: boolean}> {
```
Comment (chrisnojima) [inline shared/dev/log-send/render.native.js]: Use newer style

File: shared/dev/log-send/render.native.js [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
-  _copyToClipboard () {
+  _copyToClipboard = () => {
     NativeClipboard.setString(this.props.logSendId || '')
     this.setState({copiedToClipboard: true})
   }
 
+  _logSend = () => {
+    // We don't get the notification from the daemon so we have to do this ourselves
+    const logs = []
+    dumpLoggers((...args) => {
```
Comment (chrisnojima) [inline shared/dev/log-send/render.native.js]: When the user clicks log send we dump the period logger state and write it to a file on the device

File: shared/dev/log-send/render.native.js [hunk 4]
```diff
@@ -1,26 +1,50 @@
...
+  _copyToClipboard = () => {
     NativeClipboard.setString(this.props.logSendId || '')
     this.setState({copiedToClipboard: true})
   }
 
+  _logSend = () => {
+    if (isIOS) {
+      // We don't get the notification from the daemon so we have to do this ourselves
+      const logs = []
+      dumpLoggers((...args) => {
```
Comment (chrisnojima) [inline shared/dev/log-send/render.native.js]: get periodlogger state, write to file using RNFS

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 5]
```diff
@@ -15,9 +15,6 @@
 #import "LogSend.h"
 
 @interface AppDelegate ()
-
-@property (nonatomic, strong) LogSend * logSender;
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: turns out this doesn't work. When JS calls a native method it'll instantiate the object itself with [obj new] so it won't use our pre-existing instance. Therefore no reason to store this in app delegate

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 6]
```diff
@@ -71,12 +69,12 @@ - (void) setupGo
...
                                                    @"runmode": @"prod",
                                                    @"homedir": home,
-                                                   @"logFile": logFile,
+                                                   @"logFile": serviceLogFile,
                                                    @"serverURI": @"",
                                                    @"SecurityAccessGroupOverride": @(securityAccessGroupOverride)
                                                    } error:&err];
 
-  self.logSender = [[LogSend alloc] initWithPath:logFile];
+  [LogSend setPath:rnLogFile];
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: Instead we set this statically

File: shared/react-native/ios/Keybase/LogSend.m [hunk 7]
```diff
@@ -18,12 +16,12 @@ - (instancetype)initWithPath:(NSString *)uiLogPath {
 
   NSString *logId = nil;
   NSError *err = nil;
-  GoKeybaseLogSend(self.uiLogPath, &logId, &err);
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/LogSend.m]: was always nil

File: shared/store/action-logger.js [hunk 8]
```diff
@@ -27,6 +27,7 @@ export const actionLogger = (store: any) => (next: any) => (action: any) => {
 
   const log1 = [`Dispatching action: ${action.type}: `, forwardLogs ? JSON.stringify(actionToLog) : actionToLog]
   console.log(log1)
+  logger.log('Action:', log1)
```
Comment (chrisnojima) [inline shared/store/action-logger.js]: since we dump the period logger we also want to bookkeep the actions

File: shared/util/periodic-logger.js [hunk 9]
```diff
@@ -65,14 +65,18 @@ class PeriodicLogger {
   group () { }
   info () { }
 
-  _dump (prefix: string, args: Array<any>) {
-    console.log(prefix, ...this._logTransform(args))
+  _dump (consoleLogOverwrite: any, prefix: string, args: Array<any>) {
```
Comment (chrisnojima) [inline shared/util/periodic-logger.js]: we let dump take an override to using console.log

File: shared/util/periodic-logger.js [hunk 10]
```diff
@@ -81,25 +85,26 @@ class PeriodicLogger {
     this._messages = []
   }
 
-  dumpAll () {
-    let index = this._lastWrite
-    const endIndex = Math.max(0, index - this._size)
-    while (index >= endIndex) {
+  dumpAll (consoleLogOverwrite: any) {
```
Comment (chrisnojima) [inline shared/util/periodic-logger.js]: lets dump in order and not backwards

**Comment by chrisnojima** [comment]
- [ ] Sanity check desktop
- [ ] Branch for android version

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by MarcoPolo** [comment]
I think this might break xcode logging? At least it did for me. Maybe when you call LogSend setPath ?

File: shared/react-native/ios/Keybase/LogSend.m [hunk 11]
```diff
@@ -1,12 +1,10 @@
 #import "LogSend.h"
 
-@implementation LogSend
+NSString * logPath = @"";
```
Comment (gabriel) [inline shared/react-native/ios/Keybase/LogSend.m]: you can make this `static NSString * logPath;`?

**Comment by MarcoPolo** [comment]
nvm ^ I was wrong. :+1:

**Comment by chrisnojima** [comment]
replaced by clean pr

**Pull request closed without merging**